lib/stitches/auto_fill.py

```diff
@@ -54,18 +56,17 @@ def auto_fill(shape,
               staggers,
               skip_last,
               starting_point,
-              ending_point=None):
-    stitches = []
+              ending_point=None,
+              underpath=True):
 
-    rows_of_segments = intersect_region_with_grating(shape, angle, row_spacing, end_row_spacing)
-    segments = [segment for row in rows_of_segments for segment in row]
-
-    graph = build_graph(shape, segments, angle, row_spacing, max_stitch_length)
-    path = find_stitch_path(graph, segments, starting_point, ending_point)
+    fill_stitch_graph = build_fill_stitch_graph(shape, angle, row_spacing, end_row_spacing)
```

Comment:
This block is kind of hard to read... do you see any value in turning this into a namespace or dataclass so that you don't have to pass so many input variables around?

Reply:
I've kind of been on the fence about that.  On the one hand, it would make the function calls less cumbersome -- and I do seem to keep adding more and more parameters as time goes on.

On the other hand, I'm not sure what kind of object the class would represent.  An "autofiller"?

If you want to take a crack at it, I'd like to see what you come up with.  You're probably right that it would be tidier.  The same would apply to `lib/stitches/auto_satin.py`.

Reply:
I think for now I'm going to leave it as it is, since this PR is already getting pretty big.  I'd be open to trying your idea in a separate PR.